src/NuGet.Core/NuGet.Build.Tasks.Console/MSBuildStaticGraphRestore.cs

```diff
@@ -810,8 +810,19 @@ private PackageSpec GetPackageSpec(IMSBuildProject project, IReadOnlyDictionary<
                         .Select(s => new CompatibilityProfile(s))
                         .ToList()
                     ),
-                Version = GetProjectVersion(project)
+                Version = GetProjectVersion(project),
             };
+            string skdAnalysisLevelString = project.GetProperty("SdkAnalysisLevel");
+
+            if (skdAnalysisLevelString != null)
+            {
+                packageSpec.SdkAnalysisLevel = new NuGetVersion(skdAnalysisLevelString);
+            }
+
+            if (bool.TryParse(project.GetProperty("UsingMicrosoftNETSdk"), out bool value))
+            {
+                packageSpec.UsingMicrosoftNETSdk = value;
```

Comment:
Like other comments, with `PackageSpec.cs` now defaulting this property value to `true` (so that `JsonPackageSpecReader.cs` doesn't need to handle the default == true value), this has a bug when the project doesn't set the property to any value, and we need to treat it as false.